### DISCUSSION TITLE
Adding atomdb data to quick factories

### DISCUSSION THREAD

**Discussion by marquezj** [mctools/ncrystal#209] [Q&A]
Hi, I am trying to add new nuclides to with custom atomdb data, but I cannot find a proper solution. This is a workaround to add impurities or other minor isotopes, related to #177 

For instance, let's say I want to add He8. If I try:

`NC.load('freegas::He/He_is_He8/1perAa3;atomdb=He8:8u:0fm:1b:0b')`

then I get `NCBadInput: Invalid AtomDB specification (component "He8" is not a known element, isotope, or mixture)`. My understanding then is the atomdb data should be inside the cfg string for the factory, but I cannot find the proper syntax.

**Comment by tkittel** [comment]
You need to use:

```
NC.load('freegas::He/He8_8u_0fm_1b_0b/He_is_He8/1perAa3')
```
Anything you would normally put in the atomdb cfg-string parameter, you can put in the quick factory "name" part, but with underscores (the `He_is_He8` is exactly such as line).

For reference valid atomdb lines are described in: https://github.com/mctools/ncrystal/wiki/NCMAT-format#the-atomdb-section

**Comment by marquezj** [comment]
Perfect. OpenMC can provide the [atomic mass of the isotope](https://docs.openmc.org/en/stable/pythonapi/generated/openmc.data.atomic_mass.html) and its [free atom XS](https://docs.openmc.org/en/stable/pythonapi/generated/openmc.data.IncidentNeutron.html#openmc.data.IncidentNeutron), which can be extracted from the 0K data. This allows to add the extra nuclides in the incoherent approximation (either as `freegas` or `solid`).
